src/webui/www/private/shareratio.html

```diff
@@ -0,0 +1,168 @@
+            defaultEventType: 'keydown',
+            events: {
+                'enter': function (event) {
+                    $('save').click();
+                    event.preventDefault();
+                }
+            }
+        });
+        keyboardEvents.activate();
+
+        window.addEvent('domready', function() {
+            var hashesList = new URI().getData('hashes').split('|');
+            var origValues = new URI().getData('orig').split(',');
+
+            var values = {
+                ratioLimit: friendlyFloat(origValues[0], 2),
+                seedingTimeLimit: parseInt(origValues[1]),
+                maxRatio: friendlyFloat(origValues[2], 2),
+                maxSeedingTime: parseInt(origValues[3])
+            };
+
+            // select current values
+            if (hashesList.length === 1) {
+                var hash = hashesList[0];
+                // using double equals to compare string and int
+                if ((values.ratioLimit === UseGlobalLimit) && (values.seedingTimeLimit === UseGlobalLimit)) {
+                    // use default option
+                    setSelectedRadioValue('shareLimit', 'default');
+                }
+                else if ((values.maxRatio == NoLimit) && (values.maxSeedingTime == NoLimit)) {
+                    setSelectedRadioValue('shareLimit', 'none');
+                }
+                else {
+                    setSelectedRadioValue('shareLimit', 'custom');
+                    if (values.ratioLimit >= 0) {
+                        $('setRatio').set('checked', true);
+                        $('ratio').set('value', values.ratioLimit);
+                    }
+                    if (values.seedingTimeLimit >= 0) {
+                        $('setMinutes').set('checked', true);
+                        $('minutes').set('value', values.seedingTimeLimit);
+                    }
+                }
+            }
+
+            shareLimitChanged();
+
+            $('default').focus();
+            $('save').addEvent('click', function(e) {
+                new Event(e).stop();
+
+                if (!isFormValid()) {
+                    return false;
+                }
+
+                var shareLimit = getSelectedRadioValue('shareLimit');
+                var ratioLimitValue = 0.00;
+                var seedingTimeLimitValue = 0;
+
+                if (shareLimit === 'default') {
+                    ratioLimitValue = seedingTimeLimitValue = UseGlobalLimit;
+                }
+                else if (shareLimit === 'none') {
+                    ratioLimitValue = seedingTimeLimitValue = NoLimit;
+                }
+                else if (shareLimit === 'custom') {
+                    if ($('setRatio').get('checked'))
+                        ratioLimitValue = $('ratio').get('value');
+                    if ($('setMinutes').get('checked'))
+                        seedingTimeLimitValue = $('minutes').get('value');
+                }
+                else return false;
```

Comment:
I would add braces here to be coherent.

Reply:
Apparently, it is not accepted to write consistent code in web development, as well as adhere to any other rules... Nothing personal, it follows from all my observations of the web code.